core/src/main/java/org/infinispan/interceptors/impl/EntryWrappingInterceptor.java

```diff
@@ -823,11 +825,9 @@ protected final void wrapEntriesForPrepare(TxInvocationContext ctx, PrepareComma
          for (WriteCommand c : command.getModifications()) {
             c.setTopologyId(command.getTopologyId());
             // TODO: we need async invocation since the interceptors may do remote get
-            InvocationStage visitorStage = (InvocationStage) c.acceptVisitor(ctx, entryWrappingVisitor);
-            if (visitorStage != null) {
-               // Wait for the sub-command to finish. If there was an exception, rethrow it.
-               visitorStage.get();
-            }
+            InvocationStage visitorStage = makeStage(c.acceptVisitor(ctx, entryWrappingVisitor));
```

Comment:
```Object.wait()``` :)
Well, I think you will only have an ```InvocationStage``` when rebalance is in progress and you have to do a remote get. In 99% of the time, it should be completed when returned.